File: Leetcode/509 - Fibonacci Number/solution.js
var fib = function (n) {
    if (n <= 1) return n;
    let prev2 = 0, prev1 = 1;
    for (let i = 2; i <= n; i++) {
        [prev2, prev1] = [prev1, prev1 + prev2];
    }
    return prev1;
};

File: Leetcode/509 - Fibonacci Number/solution.py
def fib(self, n: int) -> int:
    if n <= 1:
        return n

    prev2: int = 0
    prev1: int = 1

    for _ in range(2, n + 1):
        prev2, prev1 = prev1, prev1 + prev2

    return prev1